cli/src/pcluster/validators/scheduler_plugin_validators.py

```diff
@@ -22,3 +23,67 @@ def _validate(self, grant_sudo_privileges: bool, requires_sudo_privileges: bool)
                 "but these privileges were not granted because GrantSudoPrivileges was not set or set to false.",
                 FailureLevel.ERROR,
             )
+
+
+class SchedulerPluginOsArchitectureValidator(Validator):
+    """Verify that head node architecture and os combination is supported by Scheduler Plugin."""
+
+    def _validate(self, os, architecture: str, supported_x86, supported_arm64):
+        oss_and_supported_architectures_mapping = {"x86_64": supported_x86, "arm64": supported_arm64}
+        supported_oss = oss_and_supported_architectures_mapping.get(architecture)
+        if os not in supported_oss:
+            self._add_failure(
+                "The Scheduler Plugin supports the OSs {0} for architecture {1}, none of which are "
+                "compatible with the configured OS ({2}) and the architecture supported by the head node instance "
+                "type ({1}).".format(supported_oss, architecture, os),
+                FailureLevel.ERROR,
+            )
+
+
+class SchedulerPluginRegionValidator(Validator):
+    """
+    Validate instance type and architecture combination.
+
+    Verify that head node and compute instance types imply compatible architectures.
+    """
+
+    def _validate(self, region, supported_regions):
+        if region not in supported_regions:
+            self._add_failure(
+                "The specified region {0} is not supported by Scheduler Plugin: {1}.".format(region, supported_regions),
```

Comment:
...by the scheduler plugin. Supported regions are: {}

Reply:
Changed